app/controllers/api/base_controller/generic.rb

```diff
@@ -146,14 +141,23 @@ def refresh_dialog_fields_action(dialog, refresh_fields, resource_ident)
 
       private
 
+      def collect_subcollection_data(subcollections, data)
```

Comment:
Since this modifies the data, I think we should name this something like `extract_subcollect_data!`

Reply:
I'm on board with that.  Will change.